chain/network-primitives/src/network_protocol/mod.rs

```diff
@@ -454,3 +454,24 @@ impl StateResponseInfo {
         }
     }
 }
+
+#[cfg(test)]
+mod test {
+    use std::net::IpAddr;
+    use std::net::SocketAddr;
+    use std::net::{Ipv4Addr, Ipv6Addr};
+
+    #[test]
+    fn test_from_str() {
+        use crate::types::PeerInfo;
+        use std::str::FromStr;
+        let peer_test = PeerInfo::from_str(
+            "ed25519:C6HLP37VJN1Wj2irxxZPsVsSya92Rnx12tqK3us5erKV@localhost:1337",
+        )
+        .unwrap();
+        let socket_v4 = SocketAddr::new(IpAddr::V4(Ipv4Addr::new(127, 0, 0, 1)), 1337);
```

Comment:
Use Ipv4Addr::LOCALHOST (same for Ipv6)
https://doc.rust-lang.org/std/net/struct.Ipv4Addr.html#associatedconstant.LOCALHOST